### ISSUE TITLE
Custom Handlers - Change "httpWorker" section to "custom" in host.json

### ISSUE THREAD

**Issue by anthonychu** [Azure/azure-functions-host#6080]
As discussed, let's change the `httpWorker` section to `custom` (or should it be `customHandler`?):

```json
{
    "version": "2.0",
    "customHandler": {
        "type": "http",

    }
}
```

If possible, it'd be great to support both the old and new sections for a couple of releases and output a deprecation warning at startup. If both sections are in host.json, throw an error.

/cc @pragnagopa @fabiocav @yojagad

**Comment by fabiocav** [comment]
Thank you for opening this. Yes, I think `customHandler` is appropriate.

**Comment by pragnagopa** [comment]
Flagging this as P1 and assigned to next sprint. This is a large work item as it will involve refactoring code to ensure that we have a single code base for process management between Rpc and HttpWorker layers.

**Comment by pragnagopa** [comment]
Copying notes from @anthonychu  in issue #6020 

This is great. Thanks!

I have a couple of suggestions, let me know what you think.

* Make this change *only* in the new `custom` or `customHandler` section of host.json #6080
* Change `CMD` to `command`
* If at all possible, can we come up with a way that this new method can also support this restriction? It would make it much cleaner to use and document.

    > Note: This format cannot support specifying a runtime that is part of the system path with a file extension such as 'node.exe', 'java.exe` as we cannot differentiate if user intention is relative file path or not

    For instance, could we search in the current directory first, then in the path.
* Remove the `description` section

This is what this could look like:

```json
{
    "version": "2.0",
    "customHandler": {
        "type": "http",
        "command": [ "deno", "run", "-A", "worker.bundle.js" ]
    }
}
```

**Comment by pragnagopa** [comment]
Having support for just 'command' makes it harder to figure out which argument is intended as a file vs arg that needs to be passed to the worker start up, unless we come up with a convention such as
- Starting with '-' is an arg
- Any string with a file extension is considered as a file path and we look up full path relative to function app root.

Please do let me know if you have a better suggestion.

I propose continuing to use 'description' or 'workerDescription' - for process start up information.

```json
"customHandler": {
               "type": "http",
		"workerDescription": {
                        "command": [ "deno", "run", "-A" ], // first argument is the executable. If specified takes precedence
			"defaultExecutablePath": "node",  // required if command is not specified
                         "arguments": [], // args for defaultExecutablePath. Optional
			"defaultWorkerPath": "server.js", // woker file. Optional.
                        "worerArguments": [] // args for defaultWorkerPath. Optional
		},
               "otherOptionsToConfigureHandler. For example, HttpTriggerInAndOut":"forward"
	}
```

Since we redefining options section, we can rename elements.

**Comment by anthonychu** [comment]
Summarizing a chat with @pragnagopa. There's no need to add a `command` property, as we can simplify the startup command to `defaultExecutablePath`, `arguments`, and `workingDirectory`.

```
{
    "version": "2.0",
    "customHandler": {
        "type": "http",
        "workerDescription": {
            "defaultExecutablePath": "node",  // required if command is not specified
            "arguments": [], // args for defaultExecutablePath. Optional
            "workingDirectory": "server" // absolute path, or (if possible) a relative path to /home/site/wwwroot. Optional
        },
        "httpTriggerInAndOut": "forward" // behavior for requests with only http bindings (actual name and values TBD)
    }
}
```

While it's still possible to set `defaultWorkerPath` and `workerArguments`, we leave them undocumented as we don't currently see a need for them in the context of custom handlers.

Some details:

- `defaultExecutablePath` - Can be a relative path to an executable (relative to `workingDirectory`), an executable that is in the system path, or an absolute path.
    - On Linux, this probably needs to be something like `./deno` for it to run relative to `workingDirectory`, that's okay

- `arguments` - these are arguments that are basically concatenated together into a space-delimited string

- `workingDirectory` - the folder context in which the startup command will be run. Defaults to script root (wwwroot)

For example:

```
{
    "version": "2.0",
    "customHandler": {
        "type": "http",
        "workerDescription": {
            "defaultExecutablePath": "deno.exe",
            "arguments": ["run", "-A", "worker.bundle.js"],
            "workingDirectory": "myapp"
        }
    }
}
```

Will translate into running `deno.exe run -A worker.bundle.js` with "d:\home\site\wwwroot\myapp" as the working directory.

On Linux, `defaultExecutablePath` would be `./deno` and tranlates into `./deno run -A worker.bundle.js` running in "/home/site/wwwroot/myapp".

**Comment by guibirow** [comment]
I was play around with custom handlers and noticed it does not forward the path to the handler. 
Going through the issues and PRs I discovered it will be fixed in the PR https://github.com/Azure/azure-functions-host/pull/6064 linked to this issue.

I noticed you guys are adding `"type":"http"` to consider multiple handler types according to https://github.com/Azure/azure-functions-host/issues/6061 , but you didn't account for setup multiple handlers in this configuration, so the `hosts.json` should probably look like this:

```
{
    "version": "2.0",
    "customHandler": [
        {
          "type": "http",
          "workerDescription": {
              "defaultExecutablePath": "deno.exe",
              "arguments": ["run", "-A", "worker.bundle.js"],
              "workingDirectory": "myapp"
          }
       }
    ]
}
```

Being able to define multiple `customHandlers`, will allow the use of different technologies or versions on same host. 
Let say some pieces written in rust for performance, and the others in nodejs, and so on.
